dlf/common/class.tx_dlf_solr.php

```diff
@@ -343,37 +343,39 @@ public static function solrGetCoreNumber($start = 0) {
 	 */
 	public function search($query = '') {
 
-		// Perform search.
-		$results = $this->service->search((string) $query, 0, $this->limit, $this->params);
-
-		$this->numberOfHits = count($results->response->docs);
-
 		$toplevel = array ();
 
 		$checks = array ();
 
-		// Get metadata configuration.
-		if ($this->numberOfHits > 0) {
+		// Restrict the fields to the required ones
+		$this->params['fl'] = "uid,id,toplevel";
 
-			$result = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
-				'tx_dlf_metadata.index_name AS index_name',
-				'tx_dlf_metadata',
-				'tx_dlf_metadata.is_sortable=1 AND tx_dlf_metadata.pid='.intval($this->cPid).tx_dlf_helper::whereClause('tx_dlf_metadata'),
-				'',
-				'',
-				''
-			);
+		// Get metadata configuration.
+		$result = $GLOBALS['TYPO3_DB']->exec_SELECTquery(
+			'tx_dlf_metadata.index_name AS index_name',
+			'tx_dlf_metadata',
+			'tx_dlf_metadata.is_sortable=1 AND tx_dlf_metadata.pid='.intval($this->cPid).tx_dlf_helper::whereClause('tx_dlf_metadata'),
+			'',
+			'',
+			''
+		);
 
-			$sorting = array ();
+		$sorting = array ();
 
-			while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
+		while ($resArray = $GLOBALS['TYPO3_DB']->sql_fetch_assoc($result)) {
 
-				$sorting[$resArray['index_name']] = $resArray['index_name'].'_sorting';
+			$sorting[$resArray['index_name']] = $resArray['index_name'].'_sorting';
 
-			}
+			// Add *_sorting fields to the required fields list
+			empty($this->params['fl']) ? $this->params['fl'] = $sorting[$resArray['index_name']].',' : $this->params['fl'] .= ','.$sorting[$resArray['index_name']];
```

Comment:
There should be no trailing `,` in the first case (empty `$this->params['fl']`).